### PR TITLE
Added GeoJSON export

### PR DESCRIPTION
**Pull request by rsupnekar**
Added a menu item to export features

### CODE REVIEW

File: tests/test-vandv/src/main/java/mil/emp3/test/emp3vv/navItems/AddRemoveGetTest.java [hunk 1]
```diff
@@ -145,9 +150,26 @@ public void onClick(DialogInterface dialog, int which) {
                             public void onClick(DialogInterface dialog, int which) {
                                 Log.d(TAG, "Selected Feature " + featureNames.get(which));
                                 UpdateContainer updateContainer = new UpdateContainer(activity, maps[whichMap], AddRemoveGetTest.this, styleManager);
-                                updateContainer.showUpdateContainerDialog(maps[whichMap], MapNamesUtility.getContainer(maps[whichMap],featureNames.get(which)), true);
+                                updateContainer.showUpdateContainerDialog(maps[whichMap], MapNamesUtility.getContainer(maps[whichMap], featureNames.get(which)), true);
                             }
                         }).show();
+            } else if (userAction.equals("Export features to GeoJSON")) {
+                try {
+                List<IFeature> featureList = maps[whichMap].getAllFeatures();
```
Comment (ee11cbb19052e40b07aac0ca060c23ee) [inline tests/test-vandv/src/main/java/mil/emp3/test/emp3vv/navItems/AddRemoveGetTest.java]: indentation off
> List<IFeature> featureList = maps[whichMap].getAllFeatures();

File: tests/test-vandv/src/main/java/mil/emp3/test/emp3vv/navItems/AddRemoveGetTest.java [hunk 2]
```diff
@@ -145,9 +150,26 @@ public void onClick(DialogInterface dialog, int which) {
...
+                    for (IFeature feature : featureList) {
+                        String geoJSON = GeoJsonExporter.export(feature);
+                        File file = new File("/sdcard/Download/" + feature.getName() + "Geo.json");
+                        if (!file.exists()) {
+                            boolean created = file.createNewFile();
+                        }
+                        FileOutputStream fos = new FileOutputStream(file);
```
Comment (ee11cbb19052e40b07aac0ca060c23ee) [inline tests/test-vandv/src/main/java/mil/emp3/test/emp3vv/navItems/AddRemoveGetTest.java]: add a try-with-resources statement
> FileOutputStream fos = new FileOutputStream(file);

File: tests/test-vandv/src/main/java/mil/emp3/test/emp3vv/navItems/AddRemoveGetTest.java [hunk 3]
```diff
@@ -145,9 +150,26 @@ public void onClick(DialogInterface dialog, int which) {
...
+                        fos.write(geoJSON.getBytes());
+                        fos.close();
```
Comment (ee11cbb19052e40b07aac0ca060c23ee) [inline tests/test-vandv/src/main/java/mil/emp3/test/emp3vv/navItems/AddRemoveGetTest.java]: place into finally clause
> fos.close();

File: tests/test-vandv/src/main/java/mil/emp3/test/emp3vv/navItems/AddRemoveGetTest.java [hunk 4]
```diff
@@ -145,9 +150,26 @@ public void onClick(DialogInterface dialog, int which) {
...
                             }
                         }).show();
+            } else if (userAction.equals("Export features to GeoJSON")) {
+                try {
+                List<IFeature> featureList = maps[whichMap].getAllFeatures();
+                    for (IFeature feature : featureList) {
+                        String geoJSON = GeoJsonExporter.export(feature);
+                        File file = new File("/sdcard/Download/" + feature.getName() + "Geo.json");
+                        if (!file.exists()) {
+                            boolean created = file.createNewFile();
```
Comment (ee11cbb19052e40b07aac0ca060c23ee) [inline tests/test-vandv/src/main/java/mil/emp3/test/emp3vv/navItems/AddRemoveGetTest.java]: do we care if this value is false?
> boolean created = file.createNewFile();

**Review by ee11cbb19052e40b07aac0ca060c23ee — DISMISSED** [review]

**Comment by rsupnekar** [comment]
To test, build branch and run test-vandv.  From Add Remove Get, create an overlay and some features (circle, polygon, line, etc.).  Then from the feature menu, select export features to GeoJSON.  Features exported will appear as files in /sdcard/Download/<featurename>Geo.json.

Pull the files to your PC, using adb pull.  Open each file, copy and paste the contents on this web page and validate;

http://jsonlint.com/

**Comment by mubinakhan** [comment]
Reintegrate

File: sdk/sdk-api/src/main/java/mil/emp3/json/geoJson/GeoJsonCaller.java [hunk 5]
```diff
@@ -0,0 +1,41 @@
+package mil.emp3.json.geoJson;
+
+import java.security.InvalidParameterException;
```
Comment (ee11cbb19052e40b07aac0ca060c23ee) [inline sdk/sdk-api/src/main/java/mil/emp3/json/geoJson/GeoJsonCaller.java]: Change to IllegalArgumentException
> import java.security.InvalidParameterException;

File: sdk/sdk-api/src/main/java/mil/emp3/json/geoJson/GeoJsonCaller.java [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+    static final private ICoreManager coreManager = ManagerFactory.getInstance().getCoreManager();
+
+    private GeoJsonCaller() {
+
+    }
+
+    public static void exportToString(IMap map, IFeature feature, boolean extendedData, IEmpExportToStringCallback callback) {
+
+        if ((null == feature) || (null == callback)) {
+            throw new InvalidParameterException("Parameters can't be null.");
```
Comment (ee11cbb19052e40b07aac0ca060c23ee) [inline sdk/sdk-api/src/main/java/mil/emp3/json/geoJson/GeoJsonCaller.java]: Change to IllegalArgumentException
> throw new InvalidParameterException("Parameters can't be null.");

File: sdk/sdk-api/src/main/java/mil/emp3/json/geoJson/GeoJsonCaller.java [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+        }
+
+        GeoJsonExporter exporter = new GeoJsonExporter(map, feature, extendedData, callback);
+        exporter.start();
+    }
+
+    public static void exportToString(IMap map, List<IFeature> featureList, boolean extendedData, IEmpExportToStringCallback callback) {
+
+        if ((null == featureList) || (null == callback)) {
+            throw new InvalidParameterException("Parameters can't be null.");
```
Comment (ee11cbb19052e40b07aac0ca060c23ee) [inline sdk/sdk-api/src/main/java/mil/emp3/json/geoJson/GeoJsonCaller.java]: Change to IllegalArgumentException
> throw new InvalidParameterException("Parameters can't be null.");

**Review by ee11cbb19052e40b07aac0ca060c23ee — CHANGES_REQUESTED** [review]
Minor changes:
* Change usage of InvalidParameterException to IllegalArgumentException

**Review by ee11cbb19052e40b07aac0ca060c23ee — APPROVED** [review]

**Pull request merged**